docs/user-guide/11-target-specifics.md

```diff
@@ -0,0 +1,144 @@
+
+DXC supports a few attributes and command-line arguments for targetting SPIR-V.
+You can find a document of how DXC supports [the feature mapping to SPIR-V](https://github.com/microsoft/DirectXShaderCompiler/blob/main/docs/SPIR-V.rst).
+
+Similar to DXC, Slang supports a few of the attributes as following:
+
+- [[vk::binding(binding: int, set: int = 0)]]
+    Similar to `binding` layout qualifier in Vulkan. It specifies the uniform buffer binding point, and the descriptor set for Vulkan.
+
+- [[vk::location(X)]]
+    Same as `location` layout qualifier in Vulkan. For vertex shader inputs, it specifies the number of the vertex attribute from which input values are taken. For inputs of all other shader types, the location specifies a vector number that can be used to match against outputs from a previous shader stage.
+
+- [[vk::index(Y)]]
+    Same as `index` layout qualifier in Vulkan. It is valid only when used with [[location(X)]]. For fragment shader outputs, the location and index specify the color output number and index receiving the values of the output. For outputs of all other shader stages, the location specifies a vector number that can be used to match against inputs in a subsequent shader stage.
+
+- [[vk::input_attachment_index(i)]]
+    Same as `input_attachment_input` layout qualifier in Vulkan. It selects which subpass input is being read from. It is valid only when used on subpassInput type uniform variables.
+
+- [[vk::push_constant]]
+    Same as `push_constant` layout qualifier in Vulkan. It is applicable only to a uniform block and it will be copied to a special memory location where GPU may have a more direct access to.
+
+- [vk::image_format(format : String)]
+    Same as `[[vk::image_format("XX")]]` layout qualifier in DXC. Vulkan/GLSL allows the format string to be specified without the keyword, `image_format`.  Consider the following Slang code, as an example,
+  ```csharp
+  [vk::image_format("r32f")] RWTexture2D<float> typicalTexture;
+  ```
+  It will generate the following GLSL or SPIR-V code.
+  > layout(r32f) uniform image2D typicalTexture_0;
+  > %18 = OpTypeImage %float 2D 2 0 0 2 R32f
+
+- [vk::shader_record]
+    Same as `shaderRecordEXT` layout qualifier in [GL_EXT_ray_tracing extension](https://github.com/KhronosGroup/GLSL/blob/main/extensions/ext/GLSL_EXT_ray_tracing.txt).
+    It can be used on a buffer block that represents a buffer within a shader record as defined in the Ray Tracing API.
+
+- [vk::aliased_pointer]
+    TODO
+
+- [vk_restrict_pointer]
+    TODO
+
+- [vk::spirv_instruction(op : int, set : String = "")]
+    It is similar to how `__intrinsic_op` works, but it uses the `op-code` of SPIR-V instruction set. You can also specify which instruction set you want to use as a second argument. E.G., `[[vk::spirv_instruction(1, "NonSemantic.DebugBreak")]]`
+
+- [spv_target_env_1_3]
+    TODO: I am not sure what it does. There isn't an example on slang-test
+
+
+Target specific Slang intrinsics
```

Comment:
We want to mention that ConstantBuffer translates to uniform buffer, RWByteAddressBuffer, RWStructuredBuffer, ByteAddressBuffer, StructuredBuffer translates to a shader storage buffer, where RW** buffers will have a `readwrite` layout on the buffer and the non RW ones have readonly layout.

Also want to talk about how you can control the buffer layout globally with `-fvk-use-scalar-layout`, or individually on `StructuredBuffer` with `StructuredBuffer<T, Std140Layout>` etc.

Reply:
And then we need to talk about how `ParameterBlock<T>` and `ConstantBuffer<T>` get laid out to descriptor sets.